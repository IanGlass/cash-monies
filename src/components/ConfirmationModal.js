import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
  <Modal
    isOpen={props.showModal}
    onRequestClose={props.closeModal}
    ariaHideApp={false}
    className="modal"
    contentLabel="Confirm Action"
  >
    <p>Are you sure?</p>
    <div className="button--group">
      <button className="button" onClick={props.confirmAction}>Yes</button>
      <button className="button" onClick={props.closeModal}>No</button>
    </div>
  </Modal>
);

export default ConfirmationModal;