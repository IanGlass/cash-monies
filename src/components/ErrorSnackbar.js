import React from 'react';
import { Snackbar } from '@material-ui/core';
import ErrorIcon from '@material-ui/icons/Error';


const ErrorSnackbar = (props) => (
  <Snackbar
    open={props.open}
    message={
      <div className="error-snackbar__container">
        <ErrorIcon />
        <p>{props.message}</p>      
      </div>
    }
    anchorOrigin={props.anchorOrigin}
  />
);

export default ErrorSnackbar;