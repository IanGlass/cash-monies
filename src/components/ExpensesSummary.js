import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import expensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => (
  <div>
    <p>Viewing {props.expensesCount} {props.expensesCount === 1 ? 'expense' : 'expenses'} with total {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>
  </div>
);

const mapStateToProps = (state) => ({
  expensesTotal: expensesTotal(getVisibleExpenses(state.expenses, state.filters)),
  expensesCount: getVisibleExpenses(state.expenses, state.filters).length
})

export default connect(mapStateToProps)(ExpensesSummary);