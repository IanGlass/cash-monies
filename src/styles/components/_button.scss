
.button {
  background: $blue;
  border: none;
  border-radius: 3px;
  color: white;
  font-size: $font-size-large;
  font-weight: 300;
  height: 5rem;
  padding: $s-size;
  width: 80%;
}