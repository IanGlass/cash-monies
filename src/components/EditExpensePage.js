import React from 'react';
import { connect } from 'react-redux';
import ConfirmationModal from './ConfirmationModal';
import Dialog from '@material-ui/core/Dialog';

import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  constructor() {
    super();

    this.state = {
      showModal: false,
    }
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  removeExpense = () => {
    this.closeModal();
    this.props.startRemoveExpense(this.props.expense.id);
    this.props.history.push('/');
  }
  openModal = () => {
    this.setState(() => ({
      showModal: true
    }));
  }
  closeModal = () => {
    this.setState(() => ({
      showModal: false
    }));
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary"
            onClick={this.openModal}
          >Remove Expense</button>
          <ConfirmationModal
            showModal={this.state.showModal}
            closeModal={this.closeModal}
            closeLabel="No"
            confirmLabel="Yes"
            message={"Are you sure you want to remove " + this.props.expense.description + "?"}
            confirmAction={this.removeExpense}
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense({ id }))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
