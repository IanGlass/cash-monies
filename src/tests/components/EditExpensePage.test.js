import React from 'react';
import { shallow } from 'enzyme';

import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let wrapper, history, startEditExpense, startRemoveExpense;

beforeEach(() => {
  history = { push: jest.fn() };
  startEditExpense = jest.fn();
  startRemoveExpense = jest.fn();
  wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      history={history}
      startEditExpense={startEditExpense}
      startRemoveExpense={startRemoveExpense}
    />);
});

test('should render EditExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle startEditExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  expect(wrapper).toMatchSnapshot();
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startEditExpense).toHaveBeenLastCalledWith(expenses[0].id ,expenses[0]);
});
