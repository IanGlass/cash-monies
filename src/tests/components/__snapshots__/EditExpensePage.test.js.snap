// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle editExpense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 109500,
        "createdAt": 0,
        "description": "Rent",
        "id": "1",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`should handle removeExpense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 109500,
        "createdAt": 0,
        "description": "Rent",
        "id": "1",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 109500,
        "createdAt": 0,
        "description": "Rent",
        "id": "1",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
